src/common/model/set/setLifter.ml

```diff
@@ -97,14 +97,10 @@ module Lift
     (* Check that version kinds and bars correspond to set's kind. *)
     let%lwt (bars, kind) =
       match%lwt kind s with
-      | (_, []) ->
-        add_warning WrongKind;
-        Lwt.return (32, Kind.Base.Reel) (* FIXME *)
-      | (_, [(bars, kind)]) ->
-        Lwt.return (bars, kind)
-      | (_, (bars, kind) :: _) ->
+      | Mul (_, Version (bars, kind)) -> Lwt.return (bars, kind)
+      | _ ->
         (* FIXME: more complicated that it appears *)
-        Lwt.return (bars, kind)
```

Comment:
It could be good to keep a warning here?

Reply:
I am not sure what part is more complicated here. I initially assumed it covered the previous `WrongKind` case, but looking back at it, maybe not?

Reply:
I was thinking of lines 100-101 of the previous version

Reply:
Ah, I see. No, lines 100-101 of the previous version have to do with the fact that `(_, [])` was representable but meant nothing. This case is now gone. The comment is actually kept from the previous version. It comes from sets that have a medley kind and for which checking that “the version has a compatible kind” makes little sense. I don't think there is a very good solution right now; ideally later we should check that the versions “added” as per the set's order sum up to the kind of the set, but that's more involved.

Reply:
In any case, it is a shortcoming of Dancelor, not of the sets, so I don't think we should expose a warning to the users. I suppose we could document this somewhere, either in the code some more, or open an issue or something.

Reply:
Sounds good, thanks for the explanation! Feel free to merge

Reply:
Anything we could add in the code that would make it better with respect to those considerations?

Reply:
cf https://github.com/paris-branch/dancelor/pull/232/commits/0e2fa5023128c315d4c62fbbdd30866134c164f8